js/app.js

```diff
@@ -85,11 +96,197 @@ Character.prototype.render = function() {
+		if (i < this.futureStances.length) {
+			this.futureStances[i].render(x, y);
+		}
+	}
+}
+
+
+/**
+ * Describes (a bunch about) the state that a Character is in or will probably be in in the future.
+ *
+ * Determines what actions are possible (and what the resulting next Stances will be).
+ *
+ * Can render itself on a track in a way that describes what inputs are valid and what they do.
+ *
+ * @param {Character} character - The character that the Stance describes.
+ */
+var Stance = function(character) {
+	this.character = character;
+	this.name = "abstract";
+}
+Stance.prototype.toString = function() {
+	return "Stance " + name;
+}
+
+/**
+ * Render the stance and the actions that it provides to the track.
+ *
+ * @param {number} x - The x coord of the place to render.
+ * @param {number} y - The y coord of the place to render.
+ */
+Stance.prototype.render = function(x, y) {
+	/*
+	// Draw the stance.
+    ctx.drawImage(Resources.get('images/track-stance-'+this.name+'.png'), x, y);
+
+	var offset = trackBeatLength / 3;
+	// Draw the four options for the four directions.
+    ctx.drawImage(Resources.get('images/track-'+this.spriteUp+'.png'), x, y - offset);
+    ctx.drawImage(Resources.get('images/track-'+this.spriteDown+'.png'), x, y + offset);
+    ctx.drawImage(Resources.get('images/track-'+this.spriteLeft+'.png'), x - offset, y);
+    ctx.drawImage(Resources.get('images/track-'+this.spriteRight+'.png'), x + offset, y);
+	*/
+
+	// Draw the stance.
+    ctx.fillText('stance-'+this.name, x, y);
+
+	var offset = trackBeatLength / 3;
+	// Draw the four options for the four directions.
+    ctx.fillText(this.upSprite, x, y - offset);
+    ctx.fillText(this.downSprite, x, y + offset);
+    ctx.fillText(this.leftSprite, x - offset, y);
+    ctx.fillText(this.rightSprite, x + offset, y);
+}
+
+
+/** This is a placeholder for a stance choosing the next stance upon resolving. */
+var emitNextStance = function() { /* TODO */ }
+/** This is a placeholder for a stance choosing an action. */
+var emitAction = function() { /* TODO */ }
+
+Stance.prototype.upSprite = 'stance-stumble';
+/** Do something for "up". */
+Stance.prototype.doUp = function() {
+	emitNextStance(new StumbleStance(this.character));
+}
+
+Stance.prototype.downSprite = 'stance-stumble';
+/** Do something for "down". */
+Stance.prototype.doDown = function() {
+	emitNextStance(new StumbleStance(this.character));
+}
+
+Stance.prototype.leftSprite = 'stance-stumble';
+/** Do something for "left". */
+Stance.prototype.doLeft = function() {
+	emitNextStance(new StumbleStance(this.character));
+}
+
+Stance.prototype.rightSprite = 'stance-stumble';
+/** Do something for "right". */
+Stance.prototype.doRight = function() {
+	emitNextStance(new StumbleStance(this.character));
+}
+	
+
+/**
+ * The Character is stumbling, cos they dun goofed.
+ */
+var StumbleStance = function(character) {
+	Stance.call(this, character);
+	this.name = 'stumble';
+}
+StumbleStance.prototype = new Stance(null);
```

Comment:
Does this need to run the constructor?
Is there a reason you're not using `Object.create(Stance.prototype);`?